### PR TITLE
Add button component

### LINKED ISSUE #19: [Feature] Button Component
**Is your feature request related to a problem? Please describe.**
Currently, the buttons in the library are hardcoded and lack flexibility.

**Describe the solution you'd like**
A new button component added to the library that uses Tailwind variants for customization. This would allow for more flexibility and ease of use when implementing buttons in projects.

**Additional context**
[Class Variance Authority](https://github.com/joe-bell/cva) can be used to create the tailwind variants.

### PR DESCRIPTION
**Pull request by milafreckled**
Fixes #19  

- [x] I have read and agree to the [Code of Conduct](https://github.com/eduardconstantin/azure-fundamentals/blob/main/CODE_OF_CONDUCT.md), and have followed the [Contributing guidelines](https://github.com/eduardconstantin/azure-fundamentals/blob/main/CONTRIBUTING.md).

## Pull Request Type

- [ ] Bug fix (please link related issues if applicable)
- [x] New feature
- [ ] Question update
- [ ] Other

### CODE REVIEW

File: components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            small: ['font-small', 'py-2', 'px-2']
+        },
+        compoundVariants: {
+            intent: ["primary", "secondary"],
+            size: "medium",
+        },
+        defaultVariants: {
+            intent: 'primary',
+            size: 'medium',
+            class:  ['text-white', 'rounded-lg', 'text-base/6', 'focus:outline-none', 'focus:ring-1'],
```
Comment (eduardconstantin) [inline components/Button.tsx:25]: you can move the default classes in the className prop

File: components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement>,
+VariantProps<typeof button> {}
+
+export const Button: React.FC<ButtonProps> = ({
+    className,
+    intent,
+    size,
+    ...props
+}) => <button className={button({ intent, size, className})} { ...props} />
```
Comment (eduardconstantin) [inline components/Button.tsx:38]: it should look like this
```suggestion
={`${button({ intent, size })} text-white rounded-lg focus:outline-none focus:ring-1 border mb-2 sm:mb-0`}
```

File: components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        
+        intent: {
+            primary: ['border', 'bg-blue-600/50', 'border-blue-600', 'hover:bg-emerald-600/60',
+            'focus:ring-green-800'],
+            secondary: ['bg-emerald-600/50', 'border-emerald-600', 'hover:bg-blue-600/60',
+            'focus:ring-blue-800']
+        },
+        size: {
+            large: ['font-large'],
+            medium: ['font-medium', 'text-xs', 'py-2.5', 'mb-2', 'px-5'],
```
Comment (eduardconstantin) [inline components/Button.tsx:15]:
```suggestion
            medium: ['font-medium', 'py-2.5', 'px-5', 'sm:text-sm', 'text-xs'],
```

File: components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import {cva, type VariantProps} from 'class-variance-authority'
+import React from "react"
+
+const button = cva("button", {
+    variants: {
+        
+        intent: {
+            primary: ['border', 'bg-blue-600/50', 'border-blue-600', 'hover:bg-emerald-600/60',
```
Comment (eduardconstantin) [inline components/Button.tsx:8]:
```suggestion
           primary: ["bg-blue-600/50", "border-blue-600", "hover:bg-blue-600/60", "focus:ring-blue-800", "border-blue-600"],
```

File: components/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            'focus:ring-green-800'],
+            secondary: ['bg-emerald-600/50', 'border-emerald-600', 'hover:bg-blue-600/60',
```
Comment (eduardconstantin) [inline components/Button.tsx:10]:
```suggestion
            secondary: [
        "bg-emerald-600/50",
        "border-emerald-600",
        "hover:bg-emerald-600/60",
        "focus:ring-green-800",
        "border-emerald-600",
        "sm:mr-2",
      ],
```

**Review by eduardconstantin — CHANGES_REQUESTED** [review]
Thank you the the contribution, there are a few things you need to change, also please replace the old buttons on the page.

**Comment by eduardconstantin** [comment]
@milafreckled could you resolve the other change requests?

**Comment by milafreckled** [comment]
@eduardconstantin i've just resolved, could you check if everything is ok?

File: components/Button.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            small: ['font-small', 'py-2', 'px-2']
+        },
+        compoundVariants: {
+            intent: ["primary", "secondary"],
+            size: "medium",
+        },
+        defaultVariants: {
+            intent: 'primary',
+            size: 'medium',
+            class:  ['text-white', 'rounded-lg', 'text-base/6', 'focus:outline-none', 'focus:ring-1'],
```
Comment (eduardconstantin) [inline components/Button.tsx:32]: this can be removed
```suggestion
          
```

**Review by eduardconstantin — CHANGES_REQUESTED** [review]

**Review by eduardconstantin — APPROVED** [review]

**Pull request merged**